### PR TITLE
Weekday implementation

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Function was not implemented in the interpreter.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday.txt [hunk 1]
```diff
@@ -1,8 +1,162 @@
...
+1
+
+// ******************************** STRING PARAMETERS ********************************
+
+// Invalid String passed as DateTimeValue
+>> Weekday("ABC",StartOfWeek.Sunday)
+Error({Kind:ErrorKind.InvalidArgument})
+
+// Valid date passed as string in DateTimeValue parameter
+>> Weekday(Date(2022,1,2),StartOfWeek.Sunday)
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday.txt:119]: >Date(2022,1,2) [](http://example.com/codeflow?start=11&length=14)

Should this be DateTimeValue?

Comment (Shpakh-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday.txt:119]: Also may be change date to 1/1/2023 as a boundary condition.

Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday.txt:119]: Good point, updated. Passing a string to validate coercion, and used the -01-01 to avoid ambiguity.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday_StronglyTypedEnum.txt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+>> Weekday(Date(2016, 1, 9),"")
+Errors: Error 25-27: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// Sunday, Empty string "" as StartOfWeek parameter
+>> Weekday(Date(2022,1,2),"")
+Errors: Error 23-25: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// Empty string as Date and StartOfWeek parameter
+>> Weekday("","")
+Errors: Error 11-13: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday_StronglyTypedEnum.txt:103]: also a test that passing another enum (not StartOfWeek) is erroneous?

Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday_StronglyTypedEnum.txt:108]: Added

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.
